ebean-ddl-generator/src/main/java/io/ebeaninternal/dbmigration/MChecksum.java

```diff
@@ -13,9 +15,8 @@
    * Returns the checksum of the file. Agnostic of encoding and new line character.
    */
   static int calculate(File file) {
-    try {
+    try (BufferedReader bufferedReader = IOUtils.newReader(file)) {
```

Comment:
**Important** @rbygrave This will introduce a breaking change, because we are now using UTF-8
If checksum was computed on a JVM where `Charset.defaultCharset()` is not UTF-8 (e.g. on a german windows) you may get a different checksum when the file contains special chars like "umlauts".

The risk is small, but it exists/should be mentioned in the changelog

Note: With JDK 18 https://openjdk.java.net/jeps/400, the charset will be set to UTF-8 by default, so you may get this problem anyway.

Reply:
I think this is fine in that although we generate the index with checksums it isn't being used/read yet.  The migration runner isn't actually using this checksum index file yet.

In that way we can think of this as non-breaking.